sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -977,16 +994,17 @@ private BasicDocumentRevision doGetLocalDocument(SQLDatabase db, String docId, S
 
                 return builder.buildBasicDBObjectLocalDocument();
             } else {
-                return null;
+                throw new DocumentNotFoundException("No local document found with id: " + docId + ", revId: " + revId);
             }
         } catch (SQLException e) {
-            throw new SQLRuntimeException("Error getting local document with id: " + docId, e);
+            throw new DocumentNotFoundException("Error getting local document with id: " + docId, e);
```

Comment:
Is this really a DocumentNotFoundException? Obviously something went wrong with the SQL so we can't return the document, but isn't DatastoreException more appropriate?